sdk/metric/exemplar/fixed_size_reservoir_test.go

```diff
@@ -28,7 +28,9 @@ func TestNewFixedSizeReservoirSamplingCorrectness(t *testing.T) {
 	intensity := 0.1
 	sampleSize := 1000
 
-	rng := rand.New(rand.NewSource(time.Now().UnixNano()))
+	var seed [32]byte
+	crand.Read(seed[:]) //nolint // crypto/rand never errors
+	rng := rand.New(rand.NewChaCha8(seed))
```

Comment:
Is `crypto/rand` needed here?

```suggestion
	rng := rand.New(rand.NewPCG(rand.Uint64(), rand.Uint64()))
```

Reply:
I think we can just use a default all 0 seed for a test, also makes it deterministic.

Reply:
Given this is testing the distribution of our randomness I think we want to use a varying seed to ensure that randomness is correctly checked. If we test with a single seed value we are only testing our algorithm for that specific set of pseudorandom numbers.

I do think it is a good idea to log the seed value though so if we do find an error we can check if the failure is reproducible.